packages/editor/src/components/block-drop-zone/index.js

```diff
@@ -98,7 +98,7 @@ class BlockDropZone extends Component {
 
 		if ( ! isBlockDropType( type ) ||
 			isSameBlock( srcClientId, dstClientId ) ||
-			isSrcBlockAnAncestorOfDstBlock( srcClientId, dstClientId ) ) {
+			isSrcBlockAnAncestorOfDstBlock( srcClientId, dstRootClientId ) ) {
```

Comment:
I think the logic is sound `dstClientId` is undefined when the column only contains the default appender, so the root needs to be used as a fallback.

In that case how about the following suggestion so that the previous behaviour is preserved when there is a `dstClientId`?
```suggestion
			isSrcBlockAnAncestorOfDstBlock( srcClientId, dstClientId || dstRootClientId ) ) {
```